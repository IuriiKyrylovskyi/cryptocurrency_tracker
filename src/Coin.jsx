import React from 'react';
import { BsArrowDown, BsArrowUp } from 'react-icons/bs';

const Coin = ({ coin }) => {
  const { image, name, symbol, current_price, price_change_percentage_24h } = coin;

  const diff = Math.abs(price_change_percentage_24h).toFixed(2);

  const colorClass =
    parseFloat(price_change_percentage_24h) > 0 ? 'coin__volume green' : 'coin__volume red';

  return (
    <div className="coin">
      <div className="coin__name">
        <img src={image} alt="crypto" />
        <h1>{name}</h1>
        <p className="coin__symbol">{symbol}</p>
      </div>
      <div className="coin__data">
        <p className="coin__price">{current_price}</p>
        <p className={colorClass}>
          {parseFloat(price_change_percentage_24h) > 0 ? <BsArrowUp /> : <BsArrowDown />}
          {diff} %
        </p>
      </div>
    </div>
  );
};

export default Coin;
