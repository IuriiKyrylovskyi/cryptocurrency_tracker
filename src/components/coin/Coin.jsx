import React from 'react';
import { BsArrowDown, BsArrowUp } from 'react-icons/bs';
import PropTypes from 'prop-types';

const Coin = ({ coin }) => {
  const { image, name, symbol, current_price, price_change_percentage_24h, total_volume } = coin;

  const diff = Math.abs(price_change_percentage_24h).toFixed(2);

  const colorClass =
    parseFloat(price_change_percentage_24h) > 0 ? 'coin__volume green' : 'coin__volume red';

  const coinVolumeInMillions = parseInt(total_volume / 1000000).toLocaleString();
  const price = current_price.toLocaleString();

  return (
    <div className="coin">
      <div className="coin__name">
        <img src={image} alt="crypto" />
        <h1>{name}</h1>
        <p className="coin__symbol">{symbol}</p>
      </div>
      <div className="coin__volume">$ {coinVolumeInMillions} M </div>
      <div className="coin__data">
        <p className="coin__price">$ {price}</p>
        <p className={colorClass}>
          {parseFloat(price_change_percentage_24h) > 0 ? <BsArrowUp /> : <BsArrowDown />}
          {diff} %
        </p>
      </div>
    </div>
  );
};

Coin.propTypes = {
  coin: PropTypes.object.isRequired,
  image: PropTypes.string,
  name: PropTypes.string,
  symbol: PropTypes.string,
  current_price: PropTypes.number,
  price_change_percentage_24h: PropTypes.number,
};

// Coin.defaultProps = {
//   image: 'coin image',
//   name: 'coin name',
//   symbol: '',
//   current_price: 100,
//   price_change_percentage_24h: Number(+0),
// };

export default Coin;
