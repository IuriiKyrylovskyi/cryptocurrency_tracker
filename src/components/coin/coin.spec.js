import React from 'react';
import { shallow } from 'enzyme';
import Coin from './Coin';

const setUp = props => shallow(<Coin {...props} />);

const props = {
  id: 'bitcoin',
  symbol: 'btc',
  name: 'Bitcoin',
  current_price: 41593,
  image: 'https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579',
  market_cap: 778357915130,
  market_cap_change_percentage_24h: -4.57821,
  max_supply: 21000000,
  price_change_24h: -1971.47108004,
  price_change_percentage_24h: -4.52542,
  total_supply: 21000000,
  total_volume: 141884973661,
};

let component;

describe('Coin component', () => {
  describe('should render Coin component', () => {
    beforeEach(() => {
      component = setUp();
    });

    it('should render component correctly', () => {
      expect(component).toMatchSnapshot();
    });

    it('should contain .coin wrapper', () => {
      const wrapper = component.find('.coin');

      expect(wrapper.length).toBe(1);
    });
  });

  describe('should render Coin component with props', () => {
    beforeEach(() => {
      component = setUp({ props });
    });

    it('should render .coin__volume.red correctly', () => {
      const wrapper = component.find('.coin__volume.red');
      expect(wrapper.length).toBe(1);
    });

    it('should not render .coin__volume.green', () => {
      const wrapper = component.find('.coin__volume.green');
      expect(wrapper.length).toBe(0);
    });
  });
});
