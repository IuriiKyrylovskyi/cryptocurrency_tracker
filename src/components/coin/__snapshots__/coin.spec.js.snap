// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin component should render Coin component should render component correctly 1`] = `
<div
  className="coin"
>
  <div
    className="coin__name"
  >
    <img
      alt="crypto"
      src="coin image"
    />
    <h1>
      coin name
    </h1>
    <p
      className="coin__symbol"
    />
  </div>
  <div
    className="coin__volume"
  >
    $ 
    NaN
     M 
  </div>
  <div
    className="coin__data"
  >
    <p
      className="coin__price"
    >
      $ 
      0
    </p>
    <p
      className="coin__volume red"
    >
      <BsArrowDown />
      0.00
       %
    </p>
  </div>
</div>
`;
